### PR TITLE
Fix logging of listen host

### PR DESCRIPTION
**Pull request by dolmen**
Extract the address from Addr instead of logging "0.0.0.0".

Example for `127.0.0.1:8080`:
- before: `Listening on http://0.0.0.0127.0.0.1:8080`
- after:  `Listening on http://127.0.0.1:8080`

### CODE REVIEW

File: graceful.go [hunk 1]
```diff
@@ -109,7 +110,17 @@ var (
 func LogListenAndServe(s Server, loggers ...Logger) {
 	if hs, ok := s.(*http.Server); ok {
 		logger = getLogger(loggers...)
-		logger.Printf(ListeningFormat, hs.Addr)
+		addr := hs.Addr
+		if addr == "" {
+			addr = ":80"
```
Comment (peterhellberg) [inline graceful.go]: I don't want to have a default port, especially not a restricted system port like 80, in this package. How about using ` net.SplitHostPort(hs.Addr)` directly, and have that error out if hs.Addr is empty?

Comment (dolmen) [reply graceful.go]: That will break your own tests that use ":0"...

Comment (dolmen) [reply graceful.go]: Well, `http.ListenAndServe` [defaults to ":http"](https://golang.org/src/net/http/server.go?s=80238:80279#L2627). I just tried to almost preserve this behaviour.

Comment (dolmen) [reply graceful.go]: By the way, note that listenning on `":0"` doesn't work properly either (regarding logging) as the IP stack automatically allocates a free port, so the log line is lying in that case.

Comment (peterhellberg) [reply graceful.go]: Yes, but this logging is just done as a convenience for the developer and if you ask it to listen on `:0` then you are probably aware that some random port number will be allocated.

I view this as a very unlikely edge case from what this package is meant to be used for.

**Pull request merged**